### ISSUE TITLE
Consider lazily expanding entities

### ISSUE THREAD

**Issue by ianprime0509** [ianprime0509/zig-xml#31]
Depends on #30

#30 will offer some obvious efficiency improvements for `Reader` (reducing the amount of copying necessary). A further improvement which warrants its own issue is the potential to avoid eagerly expanding entities in text content (as described in #29) and instead offering functions on `Event` to expand raw content on demand. This would avoid such processing for events where the user doesn't care about the expanded content.

This has quite a bit of overlap with the proposed design of #29, and would probably mean that `Reader` and `Parser` would use the same `Event` structure and common expansion functions.

**Comment by ianprime0509** [comment]
Implemented in the design of #36.
